src/spatialexperiment/SpatialExperiment.py

```diff
@@ -0,0 +1,633 @@
+
+        if len(self._column_pairs) > 0:
+            output += ", column_pairs=" + ut.print_truncated_dict(self._column_pairs)
+
+        if len(self._metadata) > 0:
+            output += ", metadata=" + ut.print_truncated_dict(self._metadata)
+
+        output += ")"
+        return output
+
+    def __str__(self) -> str:
+        """
+        Returns:
+            A pretty-printed string containing the contents of this object.
+        """
+        output = f"class: {type(self).__name__}\n"
+
+        output += f"dimensions: ({self.shape[0]}, {self.shape[1]})\n"
+
+        output += f"assays({len(self.assay_names)}): {ut.print_truncated_list(self.assay_names)}\n"
+
+        output += (
+            f"row_data columns({len(self._rows.column_names)}): {ut.print_truncated_list(self._rows.column_names)}\n"
+        )
+        output += f"row_names({0 if self._row_names is None else len(self._row_names)}): {' ' if self._row_names is None else ut.print_truncated_list(self._row_names)}\n"
+
+        output += (
+            f"column_data columns({len(self._cols.column_names)}): {ut.print_truncated_list(self._cols.column_names)}\n"
+        )
+        output += f"column_names({0 if self._column_names is None else len(self._column_names)}): {' ' if self._column_names is None else ut.print_truncated_list(self._column_names)}\n"
+
+        output += f"main_experiment_name: {' ' if self._main_experiment_name is None else self._main_experiment_name}\n"
+        output += f"reduced_dims({len(self.reduced_dim_names)}): {ut.print_truncated_list(self.reduced_dim_names)}\n"
+        output += f"alternative_experiments({len(self.alternative_experiment_names)}): {ut.print_truncated_list(self.alternative_experiment_names)}\n"
+        output += f"row_pairs({len(self.row_pair_names)}): {ut.print_truncated_list(self.row_pair_names)}\n"
+        output += f"column_pairs({len(self.column_pair_names)}): {ut.print_truncated_list(self.column_pair_names)}\n"
+
+        output += f"metadata({str(len(self.metadata))}): {ut.print_truncated_list(list(self.metadata.keys()), sep=' ', include_brackets=False, transform=lambda y: y)}\n"
+
+        return output
+
+    ##############################
+    #####>> spatial_coords <<#####
+    ##############################
+
+    def get_spatial_coordinates(self) -> biocframe.BiocFrame:
+        """Access spatial coordinates.
+        
+        Returns:
+            A ``BiocFrame`` containing columns of spatial coordinates.
+        """
+        return self._spatial_coords
+
+    def get_spatial_coords(self) -> biocframe.BiocFrame:
+        """Alias for :py:meth:`~get_spatial_coordinates`."""
+        return self.get_spatial_coordinates()
+
+    def set_spatial_coordinates(self, spatial_coords: biocframe.BiocFrame, in_place: bool = False) -> "SpatialExperiment":
+        """Set new spatial coordinates.
+        
+        Args:
+            spatial_coords:
+                New spatial coordinates.
+            
```

Comment:
If you update the docstring for this property in the constructor, doesn't hurt copying it again here for expected columns.